### PR TITLE
Filtering crosscorr

### PR DESCRIPTION
**Pull request by jelmerborst**
- added filtering as an option to transform_data
- some small bug fixes

### CODE REVIEW

**Comment by jelmerborst** [comment]
- added warning to filtering
- reshaped self.events to samp x trials x dim, both to remove self.data_matrix and make it in line with most other structures

File: tutorials/Workshop/dataset_raw_raw.fif
Comment (GWeindel) [inline tutorials/Workshop/dataset_raw_raw.fif:1]: Is this needed?

File: tutorials/Workshop/dataset_raw_raw_generating_events.npy
Comment (GWeindel) [inline tutorials/Workshop/dataset_raw_raw_generating_events.npy:1]: Is this needed?

File: src/hsmm_mvpy/models.py [hunk 1]
```diff
@@ -926,30 +919,27 @@ def estim_probs(self, magnitudes, parameters, n_events=None, subset_epochs=None,
...
             # computes the gains, i.e. congruence between the pattern shape
             # and the data given the magnitudes of the sensors
-            gains = gains + self.events[:,i][np.newaxis].T * magnitudes[:,i]-magnitudes[:,i]**2/2
+            for tr in range(n_trials):
+                gains[:, tr, :] =  gains[:, tr, :] + self.events[:,tr,i][np.newaxis].T * magnitudes[:,i]-magnitudes[:,i]**2/2
+        
         gains = np.exp(gains)
         probs = np.zeros([self.max_d, n_trials,n_events], dtype=np.float64) # prob per trial
         probs_b = np.zeros([self.max_d, n_trials,n_events], dtype=np.float64)# Sample and state reversed
         for trial in np.arange(n_trials):
```
Comment (GWeindel) [inline src/hsmm_mvpy/models.py:936]: This part can then be rewritten, I can take a look this afternoon

**Review by GWeindel — CHANGES_REQUESTED** [review]
Awesome! are there any performance differences?

For the gains to prob in models.py we can now make things faster by just using gains and probably invert probs to probs_b easily. I can try this afternoon

File: tutorials/Workshop/dataset_raw_raw.fif
Comment (jelmerborst) [reply tutorials/Workshop/dataset_raw_raw.fif:1]: No, don't know where this comes from.

File: tutorials/Workshop/dataset_raw_raw_generating_events.npy
Comment (jelmerborst) [reply tutorials/Workshop/dataset_raw_raw_generating_events.npy:1]: same

File: src/hsmm_mvpy/models.py (hunk 1 above)
Comment (jelmerborst) [reply src/hsmm_mvpy/models.py:936]: Yeah of course, that makes sense. Even better :)

**Comment by GWeindel** [comment]
After careful consideration we together decided to halt that development as it seems that moving from the non-sparse format of ```self.events``` (````n_samples X channels```) to a sparse format (```n_trials x max_n_samples_per_trial x channels```) yields performance issue (this branch is between 1/5th-1/10th slower than the current main). Later development might either find a way around that slow down so we close the PR but keep the branch.

**Pull request closed without merging**